metrics/lib/common.bash

```diff
@@ -267,15 +267,17 @@ function kill_processes_before_start() {
 	result=$(check_active_process "$HYPERVISOR_PATH")
 	if [[ $result -ne 0 ]]; then
 		warning "Found unexpected ${HYPERVISOR_PATH} processes present"
-		pgrep -a -f "$HYPERVISOR_PATH"
-		sudo killall "$HYPERVISOR_PATH"
+		# Sometimes we race and the process has gone by the time we list
+		# it - so make a pgrep fail non-fatal
+		pgrep -a -f "$HYPERVISOR_PATH" || true
+		sudo killall "$HYPERVISOR_PATH" || true
```

Comment:
```killall -9``` ?

Reply:
Presumably you mean as a 'stronger kill', and not a way to avoid non-0 return codes from `killall` if the process has already vanished?
I'm torn - on the one hand I'd like to be pretty sure any stray things are torn down, but not given them a chance to die semi-gracefully could leave some of our CC state in a mess, maybe.

I'm going to leave as-is for now, as I think if we start seeing a stream of things we really cannot kill then we'd like to know. At that point then I'll consider moving to something like:
```
killall $blah || true
sleep3
kill -9 $blah || true
```
OK?

Reply:
you are killing to the hypervisor, I guess you'll see a mess in the runtime, I'd prefer to use -9 (just in case the process is a zombie and ignore the term signal)

Reply:
hmmmmm - ok - I'll bite. Yes, it's the runtime that will get more messed up than anything. I'll apply `-9` to both the hypervisor and the shim code below it. Expect a re-push ...